src/backend/distributed/operations/isolate_shards.c

```diff
@@ -60,6 +62,9 @@ isolate_tenant_to_new_shard(PG_FUNCTION_ARGS)
 	Oid relationId = PG_GETARG_OID(0);
 	Datum inputDatum = PG_GETARG_DATUM(1);
 	text *cascadeOptionText = PG_GETARG_TEXT_P(2);
+	Oid shardIsolationModeOid = PG_GETARG_OID(3);
+	ShardIsolationMode shardIsolationMode = LookupShardIsolationMode(
```

Comment:
maybe just reuse LookupSplitMode instead of introducing a new type

Reply:
(this would change default behaviour, but I think that's preferable)

Reply:
I see LookupSplitMode does not properly implement "auto", and neither does this PR. The expected behaviour is that it uses non-blocking by default, but throws an error if one of the tables does not have a replica identity. We should implement that logic properly here (and we separately need to address the TODO in LookupSplitMode).